test/rules/assertions/validTypes.js

```diff
@@ -161,14 +161,82 @@ export default {
           }
       `,
       errors: [{
-        line: 3,
-        message: 'Syntax error in type: '
+        line: 2,
+        message: 'Tag @callback must have a namepath'
       }],
       settings: {
         jsdoc: {
           allowEmptyNamepaths: false
         }
       }
+    },
+    {
+      code: `
+          /**
+           * @constant {str%ng}
+           */
+           const FOO = 'foo';
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Syntax error in type: str%ng'
+        }
+      ]
+    },
+    {
+      code: `
+          /**
+           * @typedef {str%ng} UserString
+           */
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Syntax error in type: str%ng'
+        }
+      ]
+    },
+    {
+      code: `
+          /**
+           * @typedef {string} UserStr%ng
+           */
+      `,
+      errors: [
+        {
+          line: 3,
+          message: 'Syntax error in type: UserStr%ng'
```

Comment:
Can we get this to refer to a "Syntax error in namepath" instead?

Reply:
This warranted splitting `validTypeParsing` into two and a bit of duplicate code.

Also for the sake of coverage I had to remove `error.name === 'SyntaxError'` check, since `else` branch is now never executed in tests because no other kinds of errors are raised naturally. Previously on unexpected errors it would stay quiet, now it reports an error to eslint (potentially wrong if the exception was caused by mistake in code).